### PR TITLE
Docs for responsive style props

### PR DESCRIPTION
**Pull request by devongovett**
Depends on #1963.

This adds docs to the styling and layout pages for responsive style props, as well as to Provider to show how to define custom breakpoints.

### CODE REVIEW

File: packages/@react-spectrum/provider/docs/Provider.mdx [hunk 1]
```diff
@@ -76,6 +77,26 @@ Another important setting for your application is the locale. By default, React
...
+
+The Provider component can also be used to set the responsive breakpoints for your application, which are used by [layout components](layout.html#responsive-layout) and [style props](styling.html#responsive-styles) of all components within the Provider. By default, React Spectrum provides 5 breakpoints, which correspond to common device resolutions:
+
+* S – 640px
+* M - 768px
+* L – 1024px
+* XL - 1280px
+* XXL - 1536px
+
+These breakpoints can be overridden by passing the `breakpoints` prop to the `Provider` component. Any components within the Provider will consume these breakpoints and apply their style props accordingly. By default, React Spectrum uses t-shirt style names for breakpoints, but you can be name custom breakpoints however you like.
```
Comment (snowystinger) [inline packages/@react-spectrum/provider/docs/Provider.mdx:90]:
```suggestion
These breakpoints can be overridden by passing the `breakpoints` prop to the `Provider` component. Any components within the Provider will consume these breakpoints and apply their style props accordingly. By default, React Spectrum uses t-shirt style names for breakpoints, but you can rename custom breakpoints however you like.
```

**Review by snowystinger — DISMISSED** [review]
lgtm

File: packages/@react-spectrum/provider/docs/Provider.mdx [hunk 2]
```diff
@@ -76,6 +77,26 @@ Another important setting for your application is the locale. By default, React
...
+
+These breakpoints can be overridden by passing the `breakpoints` prop to the `Provider` component. Any components within the Provider will consume these breakpoints and apply their style props accordingly. By default, React Spectrum uses t-shirt style names for breakpoints, but you can name custom breakpoints however you like.
+
+This example specifies two breakpoints: `tablet` and `desktop`. It displays a [View](View.html) that changes color depending on which breakpoint is matched: green by default, blue at the `tablet` breakpoint, and magenta at the `desktop` breakpoint. See the docs for [style props](styling.html#responsive-styles) for more details.
+
+```tsx example
+<Provider theme={theme} breakpoints={{tablet: 640, desktop: 1024}}>
+  <View height="size-1000" backgroundColor={{base: 'celery-600', tablet: 'blue-600', desktop: 'magenta-600'}} />
+</Provider>
+```
```
Comment (LFDanLu) [inline packages/@react-spectrum/provider/docs/Provider.mdx:98]: I think this example is supposed to render something right? Looks like the examples on the Provider page are broken on the docs site too, this older docs build shows stuff: https://reactspectrum.blob.core.windows.net/reactspectrum/2061244489fe142e2b26da3158fdafb43fac1bba/docs/react-spectrum/Provider.html
I'll dig tmrw when I get a change

**Review by LFDanLu — CHANGES_REQUESTED** [review]
Just one thing I noticed, will take a look at why the examples are breaking in the provider docs tmrw

File: packages/@react-spectrum/provider/docs/Provider.mdx (hunk 2 above)
Comment (snowystinger) [reply packages/@react-spectrum/provider/docs/Provider.mdx:98]: sent you a slack message, we're hoping it's not broken in the prod build atm before you get sucked into it

File: packages/dev/docs/pages/react-spectrum/layout.mdx [hunk 3]
```diff
@@ -148,6 +148,91 @@ properties supported by grid.
...
+```tsx
+<Flex direction={{base: 'row', L: 'column'}} />
+```
+
+React Spectrum's breakpoints are mobile first, which means style props apply at that breakpoint and above. For example, the `L`
+breakpoint is applied at screen sizes 1024px and wider. The `base` value should be used to specify the layout at the smallest
+possible screen size, and additional breakpoints may be added to adapt the layout for larger devices.
+
+The following example shows how you could use `Grid` to define a responsive layout. It consists of a header, navigation area,
+content area, table of contents, and footer. On small screens, all of these areas are diplayed in a vertical stack, and the table of
```
Comment (dannify) [inline packages/dev/docs/pages/react-spectrum/layout.mdx:169]:
```suggestion
content area, table of contents, and footer. On small screens, all of these areas are displayed in a vertical stack, and the table of
```

**Review by dannify — CHANGES_REQUESTED** [review]

**Review by dannify — APPROVED** [review]

**Review by dannify — APPROVED** [review]

**Pull request merged**